app/controllers/rewards_controller.rb

```diff
@@ -9,6 +9,18 @@ def show
     render :show, locals: { reward: reward }
   end
 
+  def buy_reward
```

Comment:
IMO this method should go to the order controller, finally you create order here. 
A good solution is also to use transactions when you perform several operations in one method.

Reply:
Understood :) Therefore I created orders_controller and next I moved method which handles buying rewards into this controller. Thank You :)